pkg/kubectl/cmd/taint.go

```diff
@@ -248,12 +254,12 @@ func (o *TaintOptions) Complete(f cmdutil.Factory, out io.Writer, cmd *cobra.Com
 	o.builder = resource.NewBuilder(mapper, f.CategoryExpander(), typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
 		ContinueOnError().
 		NamespaceParam(namespace).DefaultNamespace()
+	if o.selector != "" {
+		o.builder = o.builder.SelectorParam(o.selector).ResourceTypes("node")
+	}
 	if o.all {
 		o.builder = o.builder.SelectAllParam(o.all).ResourceTypes("node")
 	} else {
-		if len(o.resources) < 2 {
```

Comment:
Sound good. And as you are at this, make sure one of `selector` and `all`  is present not both simultaneously if not already.